.github/workflows/test-and-deploy.yml

```diff
@@ -0,0 +1,48 @@
+name: Test and Publish Package to npmjs
+
+on: 
+  release:
+    types: [created]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Setup Node.js
+      uses:  actions/setup-node@v3
+      with:
+        node-version: '10'
+      
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Run lint
+      run: npm run lint
+    
+    - name: Run test
+      run: npm run test
+    
+    - name: Run build
+      run: npm run build
+
+  deploy:
+    if: success()
```

Comment:
Se entendi certo, esse deploy vai acontecer sempre e não só nas tag, ou tô enganado?

Reply:
Eu coloquei para rodar o workflow somente quando uma release é criada: 
```
on: 
  release:
    types: [created]
```
esse if: success() retorna true se nenhum passo anterior falhar:

"success
Returns true when none of the previous steps have failed or been canceled."